components/camera/server.go

```diff
@@ -54,16 +57,22 @@ func (s *serviceServer) GetImage(
 
 	// Determine the mimeType we should try to use based on camera properties
 	if req.MimeType == "" {
-		if _, ok := s.imgTypes[req.Name]; !ok {
+		s.imgTypesMu.RLock()
+		imgType, ok := s.imgTypes[req.Name]
+		s.imgTypesMu.RUnlock()
+		if !ok {
 			props, err := cam.Properties(ctx)
```

Comment:
Thanks for keeping the lock scoped to exactly the map accesses, that's smart. I fear that `Properties` call can be blocking.